lib/ukrandom/dev.c

```diff
@@ -44,18 +45,46 @@
 
 #define DEV_RANDOM_NAME "random"
 #define DEV_URANDOM_NAME "urandom"
+#define DEV_HWRNG_NAME "hwrng"
+
+/* TODO: Move this to random.c once ukarch_random is implemented as a driver */
+__ssz uk_random_hwrng_fill_buffer(void *buf, __sz buflen)
+{
+	__sz step, chunk_size, i;
+	__u64 rd;
+	int ret;
+
+	step = sizeof(__u64);
+	chunk_size = buflen % step;
+
+	for (i = 0; i < buflen - chunk_size; i += step) {
+		ret = ukarch_random_seed_u64((__u64 *)((char *)buf + i));
+		if (unlikely(ret))
+			return ret;
+	}
+
+	/* fill the remaining bytes of the buffer */
+	if (chunk_size > 0) {
+		ret = ukarch_random_seed_u64(&rd);
+		if (unlikely(ret))
+			return ret;
+		memcpy((char *)buf + i, &rd, chunk_size);
+	}
+
+	return buflen;
+}
 
 static int dev_random_read(struct device *dev __unused, struct uio *uio,
 			   int flags __unused)
 {
-	size_t count;
+	__sz count;
 	char *buf;
 
 	buf = uio->uio_iov->iov_base;
 	count = uio->uio_iov->iov_len;
 
-	uk_swrand_fill_buffer(buf, count);
-
+	uk_random_fill_buffer(buf, count);
+	
```

Comment:
Stray tabs here. This should had been caught by the checkpatch check, yet it did not. It seems to be related to govctl as the issue is properly reported if I run `checkpatch.uk` against that commit from the command-line.

CC: @craciunoiuc